### PR TITLE
 Project list API

### PR DESCRIPTION
**Pull request by LittleSound**

### CODE REVIEW

File: Controllers/IncubatorController.cs [hunk 1]
```diff
@@ -26,5 +29,66 @@ CancellationToken cancellationToken
             var kbyInfo = tokenRepository.GetSingle("KBY");
             return ApiResult(kbyInfo);
         }
+
+        [HttpGet("list")]
+        [ProducesResponseType(typeof(ApiResult<IEnumerable<ProjectManifest>>), 200)]
+        [ProducesResponseType(typeof(ApiResult), 404)]
+        public async Task<IActionResult> HomeProjectList(
```
Comment (IsQiao) [inline Controllers/IncubatorController.cs]: rename the action name is "List"

File: Controllers/IncubatorController.cs [hunk 2]
```diff
@@ -26,5 +29,66 @@ CancellationToken cancellationToken
...
+            GetProjectListRequest request,
+            [FromServices] KyubeyContext db,
+            [FromServices] TokenRepositoryFactory tokenRepositoryFactory,
+            CancellationToken cancellationToken
+            )
+        {
+            //var tokens = await db.Tokens.ToListAsync();
+            var tokenRepository = await tokenRepositoryFactory.CreateAsync(request.Lang);
+            var kbyInfo = tokenRepository.EnumerateAll();
```
Comment (IsQiao) [inline Controllers/IncubatorController.cs]: kbyInfo?

File: Controllers/IncubatorController.cs [hunk 3]
```diff
@@ -26,5 +29,66 @@ CancellationToken cancellationToken
...
+        [ProducesResponseType(typeof(ApiResult<IEnumerable<ProjectManifest>>), 200)]
+        [ProducesResponseType(typeof(ApiResult), 404)]
+        public async Task<IActionResult> HomeProjectList(
+            GetProjectListRequest request,
+            [FromServices] KyubeyContext db,
+            [FromServices] TokenRepositoryFactory tokenRepositoryFactory,
+            CancellationToken cancellationToken
+            )
+        {
+            //var tokens = await db.Tokens.ToListAsync();
```
Comment (IsQiao) [inline Controllers/IncubatorController.cs]: remove the comments

File: Controllers/IncubatorController.cs [hunk 4]
```diff
@@ -26,5 +29,66 @@ CancellationToken cancellationToken
...
+            var tokenRepository = await tokenRepositoryFactory.CreateAsync(request.Lang);
+            var kbyInfo = tokenRepository.EnumerateAll();
+            var ProjectManifests = new List<ProjectManifest>();
+
+            kbyInfo = kbyInfo.Where(x => x?.Incubation != null).ToList().Skip(request.Skip).Take(request.Take);
+            var dbIncubations = await db.Tokens.Where(x =>
+                x.HasIncubation && kbyInfo.FirstOrDefault(t => x.Id == t.Id).Incubation != null &&
+                x.Status == TokenStatus.Active).ToListAsync();
+
+            switch (request.status)
```
Comment (IsQiao) [inline Controllers/IncubatorController.cs]: please rename them ,like 'not_started','in_progress','over' or 0,1,2

File: Controllers/IncubatorController.cs [hunk 5]
```diff
@@ -26,5 +29,66 @@ CancellationToken cancellationToken
...
+
+            switch (request.ranking)
+            {
+                case "latest":
+                    kbyInfo = kbyInfo.OrderByDescending(x => (x.Incubation.Begin_Time ?? DateTime.MinValue));
+                    break;
+                case "money":
+                    kbyInfo = kbyInfo.OrderByDescending(x => dbIncubations.FirstOrDefault(s => s.Id == x.Id)?.Raised ?? 0);
+                    break;
+                case "comment":
```
Comment (IsQiao) [inline Controllers/IncubatorController.cs]: remove

File: Models/ProjectManifest.cs [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+        public string Cover { get; set; }
+
+        public string Introduction { get; set; }
+
+        public DateTime? Start_Time { get; set; }
+
+        public DateTime DeadLine { get; set; }
+
+        public decimal Target_Amount { get; set; }
```
Comment (IsQiao) [inline Models/ProjectManifest.cs]: rename it ''TargetAmount"

File: Models/ProjectManifest.cs [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+{
+    public class ProjectManifest
+    {
+        public string Id { get; set; }
+
+        public string Cover { get; set; }
+
+        public string Introduction { get; set; }
+
+        public DateTime? Start_Time { get; set; }
```
Comment (IsQiao) [inline Models/ProjectManifest.cs]: "StartTime"

File: Models/ProjectManifest.cs [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+        public DateTime DeadLine { get; set; }
+
+        public decimal Target_Amount { get; set; }
+
+        public decimal Target_Credits { get; set; }
```
Comment (IsQiao) [inline Models/ProjectManifest.cs]: rename

File: Models/ProjectManifest.cs [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+        public int Number_Of_Supporters { get; set; }
```
Comment (IsQiao) [inline Models/ProjectManifest.cs]: remove "_"

File: Controllers/IncubatorController.cs [hunk 10]
```diff
@@ -26,5 +29,66 @@ CancellationToken cancellationToken
...
+            GetProjectListRequest request,
+            [FromServices] KyubeyContext db,
+            [FromServices] TokenRepositoryFactory tokenRepositoryFactory,
+            CancellationToken cancellationToken
+            )
+        {
+            //var tokens = await db.Tokens.ToListAsync();
+            var tokenRepository = await tokenRepositoryFactory.CreateAsync(request.Lang);
+            var kbyInfo = tokenRepository.EnumerateAll();
+            var ProjectManifests = new List<ProjectManifest>();
```
Comment (IsQiao) [inline Controllers/IncubatorController.cs]: Rename The model name is "GetIncubatorListResponse"

Comment (IsQiao) [inline Controllers/IncubatorController.cs]: Variable names start in lowercase

**Review by LittleSound — COMMENTED** [review]
correct

**Pull request merged**